src/Engine/Objects/CombinedSkillValue.h

```diff
@@ -110,4 +111,12 @@ class CombinedSkillValue {
 
     static bool isLevelValid(int level);
     static bool isMasteryValid(PLAYER_SKILL_MASTERY mastery);
+
+    explicit operator bool() const { return _level > 0; }
+    void learn();
+    void reset();
+
+    void subtract(const int sub);
+    void add(const int add);
+    void set(const int set);
```

Comment:
I suggest we introduce a static factory method: 

``` 
CombinedSkillValue CombinedSkillValue::fromJoined(uint16_t)
```

And then just use assignment. And also drop the `explicit CombinedSkillValue(int joinedValue)` in favor of the factory method.